bluemira/power_cycle/base.py

```diff
@@ -0,0 +1,468 @@
+        delimiters.
+        If the input is a `list`, elements of the list are considered
+        the valid values. If the input is a `dict`, the dictionary keys
+        are considered the valid values.
+        """
+        if isinstance(valid_values, dict):
+            values_list = list(valid_values.keys())
+        elif isinstance(valid_values, list):
+            values_list = valid_values
+        else:
+            raise TypeError(
+                """
+                The argument to be transformed into a message of valid
+                values is not a `list`.
+                """
+            )
+
+        # Convert every value to string
+        string_values = [str(element) for element in values_list]
+
+        # Create string message
+        values_msg = "', '".join(string_values)
+        values_msg = "'" + values_msg + "'"
+
+        # Output message
+        return values_msg
+
+    @staticmethod
+    def add_dict_entries(dictionary, new_entries):
+        """
+        Add (key,value) pairs to a dictionary, only if they are not
+        already specified (i.e. no substitutions). If dictionary is
+        empty, returns only `new_entries`.
+
+        Parameters
+        ----------
+        dictionary: `dict`
+            Dictionary to be modified.
+        new_entries: `dict`
+            Second dictionary, which entries will be added to
+            `dictionary`, unless they already exist.
+
+        Returns
+        -------
+        dictionary: `dict`
+            Modified dictionary.
+        """
+
+        # Validate whether `dictionary` exists (i.e. not empty)
+        if dictionary:
+
+            # Keys of new entries
+            new_entries_keys = new_entries.keys()
+
+            # For each key
+            for key in new_entries_keys:
+
+                # Current entry value
+                value = new_entries[key]
+
+                # Add entry to dictionary, if not yet there
+                dictionary.setdefault(key, value)
+        else:
+
+            # For empty `dictionary`, output only `new_entries`
+            dictionary = new_entries
+
+        # Output extended dictionary
+        return dictionary
+
+    # ------------------------------------------------------------------
+    # PLOT MANIPULATION
+    # ------------------------------------------------------------------
+    @staticmethod
+    def validate_axes(ax):
+        """
+        Validate axes argument for plotting method. If `None`, create
+        new `axes` instance.
+        """
+        if ax is None:
+            ax = plt.gca()
+        elif not isinstance(ax, plt.Axes):
+            raise TypeError(
+                """
+                The argument 'ax' used to create a plot is not
+                an instance of the `Axes` class.
+                """
+            )
+        return ax
+
+    @classmethod
+    def adjust_2d_graph_ranges(cls, x_frac=0.1, y_frac=0.1, ax=None):
```

Comment:
not a classmethod

Reply:
@je-cook,
I have defined this as a classmethod because I call another static method from the same class inside the routine.
Using a classmethod ensures the implementation is not dependent on the name I give to the class. Is this wrong?

Reply:
That is true but is not the whole story. As I said above I would only really use a `classmethod` to modify the class or possibly to modify initialisation of the class neither are the case here. 

In fact I would suggest that this function should be a `staticmethod` or possibly a pure function outside the class as all it is doing is creating an axis. I would probably call the validation before I create some axes (in the method that calls this function).

This function will always (if designed in the expected way) be called from a class instance. If you want to both keep the method in the class and internally call the validate method (the only use of `cls`) then this is just a normal method because it requires access to the accessible methods within the class.

Reply:
I have resolved this problem in this particular case by doing what @hsaunders1904 suggested and placing both methods as stand-alone functions in a separate file 'tools.py'. I will check if this issue comes up anywhere else.